File: deleteBeforeCollonRunSub.py
def process_input(input_text):
    # Split input_text into lines
    lines = input_text.strip().split('\n')
    
    # Define mappings for replacements
    replacements = {
        '[Status_Chamada]': '{global:tipoDesconexao}',
        '[Ultimo_Ponto]': '{global:ultimoPonto}',
        '[Ultimo_Ponto_Da_Chamada]': '{global:ultimoPonto}',
        '[Ultimo_Ponto_Fluxo_Anterior]': '{ultimoPontoFluxoAnterior}',
        '[Fluxo_Anterior]': '{global:fluxoAnterior}',
        '[Input_Cliente]': '{MRES}',
        '[Status_Code]': '{global:statusCode}',
        '[Produto]': '{global:Tag_produto}',
        'Timestamp:': ''  # This will delete the entire line containing 'Timestamp:'
    }
    
    processed_lines = []
    
    for line in lines:
        # Remove leading and trailing whitespace
        line = line.strip()
        
        # Check if the line contains 'Timestamp:' to delete it
        if 'Timestamp:' in line:
            continue  # Skip this line
        
        # Remove the label (part before the colon :)
        # Implement this later
            #characters = [':', ' ']
            #colon_index = min(line.find(char) for char in characters)
        colon_index = line.find(':')
        if colon_index != -1:
            line = line[colon_index + 1:].strip()
        
        # Replace [Status_Chamada] and [Ultimo_Ponto] with {global}
        for key, value in replacements.items():
            if key in line:
                line = line.replace(key, value)
        
        # Append the processed line
        processed_lines.append(line)
    
    # Join processed lines into a single string with newlines
    processed_text = '\n'.join(processed_lines)
    
    return processed_text

# Example usage:
input_text = """
event_name: perfil_cliente
Validacao: Segurado
Fluxo: Menu_Chutes
Indicador: [Status_Chamada]
Range: 16000
Ponto: 16011
Ultimo_Ponto: [Ultimo_Ponto]
Input:
Tentativa:
Ponto_Origem:
Produto: Cobranca_Seguros
Timestamp: [Time_Stamp]
"""

output_text = process_input(input_text)
print(output_text)

File: deleteBeforeCollonRunSub.js
// Function to process the HTML content
function processHtmlPage(htmlContent) {
    // Create a temporary div element to hold the HTML content
    var tempDiv = document.createElement('div');
    tempDiv.innerHTML = htmlContent.trim();
    
    // Define mappings for replacements
    var replacements = {
        '[Status_Chamada]': '{global}',
        '[Ultimo_Ponto]': '{global}',
        'Timestamp:': ''  // This will delete the entire line containing 'Timestamp:'
    };

    // Select all <p> tags (you can adjust for other tags as needed)
    var paragraphs = tempDiv.querySelectorAll('p');

    // Array to hold processed lines
    var processedLines = [];

    paragraphs.forEach(function(p) {
        // Get the text content of the paragraph
        var line = p.textContent.trim();

        // Check if the line contains 'Timestamp:' to delete it
        if (line.includes('Timestamp:')) {
            return;  // Skip this line
        }

        // Remove the label (part before the colon :)
        var colonIndex = line.indexOf(':');
        if (colonIndex !== -1) {
            line = line.substring(colonIndex + 1).trim();
        }

        // Replace [Status_Chamada] and [Ultimo_Ponto] with {global}
        Object.entries(replacements).forEach(function([key, value]) {
            line = line.replace(key, value);
        });

        // Add the processed line to the array
        processedLines.push(line);
    });

    // Join processed lines into a single string with newlines
    var processedHtml = processedLines.join('<br>');

    // Output or use the processed HTML content as needed
    console.log(processedHtml);
    // If you want to update an existing element with the processed content:
    // document.getElementById('output').innerHTML = processedHtml;
}

// Example usage:
var htmlContent = `
<div>
    <p>event_name:cliente_identificado</p>
    <p>Validacao:Identificado</p>
    <p>Fluxo:Ura_Cdc_Inicio_Topo_Ura</p>
    <p>Indicador: [Status_Chamada]</p>
    <p>Range:10500</p>
    <p>Ponto:10508</p>
    <p>Ultimo_Ponto:[Ultimo_Ponto]</p>
    <p>Input:</p>
    <p>Tentativa:</p>
    <p>Ponto_Origem:</p>
    <p>Produto:Cdc</p>
    <p>Timestamp:[Time_Stamp]</p>
</div>
`;

// Call the function with the example HTML content
processHtmlPage(htmlContent);
